mutation_partition.cc

```diff
@@ -1544,6 +1544,23 @@ mutation_partition::live_row_count(const schema& s, gc_clock::time_point query_t
     return count;
 }
 
+uint64_t
+mutation_partition::expired_row_count(const schema& s, gc_clock::time_point gc_before, gc_clock::time_point query_time) const {
+    check_schema(s);
+    uint64_t count = 0;
+
+    for (const rows_entry& e : non_dummy_rows()) {
+        tombstone base_tombstone = range_tombstone_for_row(s, e.key());
+        if (!e.row().is_live(s, column_kind::regular_column, base_tombstone, query_time)) {
+            if (e.row().deleted_at().max_deletion_time() < gc_before) {
+                ++count;
+            }
+        }
+    }
+
+    return count;
```

Comment:
This will take a long time, and cause a reactor stall, for large partitions.

Reply:
is there a reason to use maybe_yield() inside of cycle here? either at higher level?
I suspected should do something like that but used _mutation_partition::live_row_count_ as example, would follow your recomendations

Reply:
You can yield either in a function that returns a future, or in a seastar::thread. So if you want to yield here, you must be sure the function is running in a seastar::thread.